### PR TITLE
CT-3165 specify complaint type, subtype, priority

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
This is a small PR adding the fields and values for the complaint type, subtype and priority. There's a hack in the offender_complaint to pre-fill the fields since there is no UI to do this yet, this will be removed in the follow up PR

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
n/a
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-3165

### Deployment
Adds a data migration task to add default values for the new fields so that any existing complaint cases can still be saved and worked with

### Manual testing instructions
none

### CODE REVIEW

File: app/models/case/sar/offender_complaint.rb [hunk 1]
```diff
@@ -5,6 +5,43 @@ class Case::SAR::OffenderComplaint < Case::SAR::Offender
...
+
+  enum complaint_subtype: {
+    missing_data:  'missing_data',
+    inaccurate_data: 'inaccurate_data',
+    redacted_data: 'redacted_data',
+    timeliness: 'timeliness',
+  }
+
+  enum priority: {
+    normal:  'normal',
```
Comment (willmcb) [inline app/models/case/sar/offender_complaint.rb:31]: Doesn't priority already exist? Not sure there is a need to have 'normal' at all given that is the default. Is it not just a boolean flag to say something is high priority 🤔

File: app/models/case/sar/offender_complaint.rb [hunk 2]
```diff
@@ -5,6 +5,43 @@ class Case::SAR::OffenderComplaint < Case::SAR::Offender
...
+    high: 'high',
+  }
+
+  # CT-3165 WIP REQUIRED FOR VALIDATIONS
+  #         REMOVE ONCE UX IS COMPLETED
+  before_validation :set_types
```
Comment (willmcb) [inline app/models/case/sar/offender_complaint.rb:37]: 👍

File: db/data_migrations/20201126094009_set_types_for_complaints.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class SetTypesForComplaints < ActiveRecord::DataMigration
+  def up
+    kases = Case::Base.offender_sar_complaint.all
```
Comment (willmcb) [inline db/data_migrations/20201126094009_set_types_for_complaints.rb:3]: Is this best as a migration or a rake task? Just thinking about the sequencing of the migration. Maybe I have the wrong end of the stick, but could you explain your reasoning to me :)

File: spec/models/case/sar/offender_complaint_spec.rb [hunk 4]
```diff
@@ -44,6 +44,87 @@
...
+          build(:offender_sar_complaint, complaint_type: 'wibble')
+        }.to raise_error ArgumentError
+      end
+    end
+  end
+
+  describe '#priority' do
+    context 'validates that priority is not blank' do
+      it 'errors' do
+        pending 'removal of the "set_type" validation hack'
```
Comment (willmcb) [inline spec/models/case/sar/offender_complaint_spec.rb:77]: Nice

**Review by willmcb — COMMENTED** [review]
Can we have a quick chat?

**Review by ymao2 — APPROVED** [review]

File: app/models/case/sar/offender_complaint.rb (hunk 1 above)
Comment (Polsonby) [reply app/models/case/sar/offender_complaint.rb:31]: Great point... 🤔 I guess I've approached it from the point of view that there are really three priorities but we're collapsing them into two. Using the enum even with a boolean automatically gives us a full set of methods we can use to query and set the value, and using the strings makes it easier if we end up extending this beyond two states one day.

File: db/data_migrations/20201126094009_set_types_for_complaints.rb (hunk 3 above)
Comment (Polsonby) [reply db/data_migrations/20201126094009_set_types_for_complaints.rb:3]: You're quite right - I did consider making it a rake task but we never remember to run them. Happy to change if you like?

**Pull request merged**